File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string word;
        string reverseWord;
        for (char c : s) {
            if (isalnum(c) != 0) {
                word += tolower(c);
            } 
        }
        for (int i = word.length() - 1; i >= 0; i--) {
            reverseWord += word[i];
        }
        return word == reverseWord;
    }
};

File: 125-valid-palindrome/valid-palindrome.go
func isPalindrome(s string) bool {
    left := 0
    right := len(s) - 1
    for left < right {
        for left < right && !isAlphanumeric(rune(s[left])) {
            left++
        }
        for left < right && !isAlphanumeric(rune(s[right])) {
            right--
        }
        
        if strings.ToLower(string(s[left])) != strings.ToLower(string(s[right])) {
            return false
        }

        left++
        right--
    }
    return true   
}

func isAlphanumeric(r rune) bool {
    return unicode.IsLetter(r) || unicode.IsDigit(r)
}